src/Extensions/Abstractions/ExecutionContext/BaseExecutionContext.cs

```diff
@@ -52,6 +55,22 @@ public BaseExecutionContext(IHostEnvironment? hostEnvironment = null)
 			}
 
 			IsCanary = false;
+
+			// Service Fabric specific values
+			ServiceName = GetVariable(ServiceNameVariableName) ?? DefaultEmptyValue;
+			ApplicationName = GetVariable(ApplicationNameVariableName) ?? DefaultEmptyValue;
+
+			string nodeName = GetVariable(NodeNameVariableName) ?? DefaultEmptyValue;
+			MachineId = FormattableString.Invariant($"{MachineName}_{nodeName}");
+
+			string? nodeIPAddressOrFQDN = GetVariable(NodeIPOrFQDNVariableName);
+
+			if (IPAddress.TryParse(nodeIPAddressOrFQDN, out IPAddress? ipAddress))
+			{
+				ClusterIpAddress = ipAddress;
+			}
+
+			Cluster = GetVariable(ClusterNameVariableName) ?? nodeIPAddressOrFQDN ?? MachineId;
```

Comment:
So if cluster name is not specified, it will be different for each replica/instance?

Reply:
That's to copy existing behavior, what alternative would you suggest?

Reply:
Then existing behavior is wrong. Leave it empty if not specified.